docs/capi-vsphere.md

```diff
@@ -0,0 +1,178 @@
+NODE_IPAM_POOL_KIND: "InClusterIPPool"
+NAMESERVER: "8.8.8.8"
+providers:
+  - name: incluster
+    url: https://github.com/kubernetes-sigs/cluster-api-ipam-provider-in-cluster/releases/latest/ipam-components.yaml
+    type: IPAMProvider
+
+```
+### Management cluster initialization
+
+To initialize the management cluster with vSphere infrastrcture provider and additional IPAM provider you can run:
+
+```
+clusterctl init --infrastructure vsphere --ipam incluster
+```
+
+For more details on Cluster API Provider vSphere see it's [docs](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/tree/main/docs).
+
+### IPAM IP pool creation
+
+1. Define IP pool that will be used to assign IP addresses to a child cluster machines in `capi-ipam.yaml`:
+
+```yaml
+apiVersion: ipam.cluster.x-k8s.io/v1alpha1
+kind: InClusterIPPool
+metadata:
+  name: ipam-ip-pool
+spec:
+  subnet: 192.168.117.0/24
+  gateway: 192.168.117.1
+  start: 192.168.117.152
+  end: 192.168.117.180
+```
+
+2. Deploy the IPAM configuration: 
+```
+kubectl apply -f capi-ipam.yaml
+```
+
+## MetalLB as Load Balancer solution
+
+Classic vSphere does not have a built-in solution that can be leveraged on Kubernetes level to create LoadBalancer service type. In order to make it possible, one of the common solution is to use MetalLB (please find it's docs [here](https://metallb.universe.tf/))
+
+1. Configure MetalLB IP address pool which will be used to assing IP addresses from (please adapt it for your network):
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: example-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - 192.168.10.100-192.168.10.150
+```
+
+2. Create MetalLB L2advertisement object:
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  annotations:
+  name: example-l2-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+  - example-pool
+```
+
+## Child cluster deployment
+
+Once all the controllers are up and running, you can apply the cluster manifests containing the specifications of the cluster you want to provision.
+
+
+### Generate a child cluster definition using the template:
+
+#### For control plane in Pods:
+
+If you want to have child cluster control plane in a Pods, use [this template](capi-vsphere-tmpl-cp-in-pods.yaml): 
+
+```sh
+clusterctl generate cluster my-cluster --control-plane-machine-count 1 --worker-machine-count 1 --from k0scluster-tmpl-cp-in-pods.yaml > my-cluster.yaml
+```
+
+#### For control plane in VMs:
+
+If you want to have child cluster control plane in separate VMs, use [that template](capi-vsphere-tmpl-cp-in-pods.yaml) (CP will consist of 3 controllers):
+
+```sh
+clusterctl generate cluster my-cluster --worker-machine-count 1 --from k0scluster-tmpl-cp-in-vms.yaml > my-cluster.yaml 
+```
+
+### Deploy the child clusters
+
+1. Deploy the child cluster: `kubectl apply -f my-cluster.yaml`. Check the deployment status in the Mothership.
```

Comment:
I would suggest to say: "You can check the deployment status in the management cluster."
And not "Check the deployment status in the Mothership."